tests/integration/test_dynamic_sampling.py

```diff
@@ -609,20 +624,43 @@ def assert_client_reports():
     SAMPLE_RATE = 0.001
     _add_sampling_config(config, sample_rate=SAMPLE_RATE, rule_type=rule_type)
 
-    envelope, trace_id, event_id = event_factory(
-        public_key, client_sample_rate=SAMPLE_RATE
-    )
+    NUM_EVENTS = 200
 
-    relay.send_envelope(project_id, envelope)
-    assert_client_reports()
+    # Test 1: Send events with client_sample_rate=SAMPLE_RATE
+    # The server should adjust sampling to let most of these through
+    # since the client already did the sampling
+    for i in range(NUM_EVENTS):
+        envelope, trace_id, event_id = event_factory(
+            public_key, client_sample_rate=SAMPLE_RATE
+        )
+        relay.send_envelope(project_id, envelope)
 
-    envelope, trace_id, event_id = event_factory(public_key, client_sample_rate=1.0)
+    accepted_with_client_sampling, client_reports_1 = collect_events_batch(NUM_EVENTS)
 
-    relay.send_envelope(project_id, envelope)
-    assert_client_reports()
+    # Test 2: Send events with client_sample_rate=1.0
+    # The server should apply full sampling rate, dropping most of the event.
+    for i in range(NUM_EVENTS):
+        envelope, trace_id, event_id = event_factory(public_key, client_sample_rate=1.0)
+        relay.send_envelope(project_id, envelope)
 
-    with pytest.raises(queue.Empty):
-        mini_sentry.captured_events.get(timeout=1)
+    accepted_without_client_sampling, client_reports_2 = collect_events_batch(
+        NUM_EVENTS
+    )
+
+    # Both should drop most events due to the 0.001 sample rate
+    total_events = NUM_EVENTS * 2
+    total_accepted = accepted_with_client_sampling + accepted_without_client_sampling
+
+    # Expect to drop most of the events (allowing some variance)
+    assert (
+        total_accepted <= total_events * 0.01
+    ), f"Expected ~99% drop rate overall, got {total_accepted}/{total_events} accepted"
```

Comment:
You're using 0.01/99% here, but `SAMPLE_RATE = 0.001`. This is off by a decimal place, I assume that's a typo?

Reply:
I actually wanna allow more variance and actually still allow upto 1% through without failing the assert. 
Do you think it's not a good idea?

Reply:
No, I think that makes sense. It could maybe use a comment, since this way it really does look like typo IMO.